packages/flutter/test/material/switch_list_tile_test.dart

```diff
@@ -618,4 +618,117 @@ void main() {
           )
     );
   });
+
+  testWidgets('SwitchListTile respects trackOutlineColor in active/enabled states', (WidgetTester tester) async {
+    const Color activeEnabledTrackOutlineColor = Color(0xFF000001);
+    const Color activeDisabledTrackOutlineColor = Color(0xFF000002);
+    const Color inactiveEnabledTrackOutlineColor = Color(0xFF000003);
+    const Color inactiveDisabledTrackOutlineColor = Color(0xFF000004);
+
+    Color getOutlineColor(Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        if (states.contains(MaterialState.selected)) {
+          return activeDisabledTrackOutlineColor;
+        }
+        return inactiveDisabledTrackOutlineColor;
+      }
+      if (states.contains(MaterialState.selected)) {
+        return activeEnabledTrackOutlineColor;
+      }
+      return inactiveEnabledTrackOutlineColor;
+    }
+
+    final MaterialStateProperty<Color> trackOutlineColor = MaterialStateColor.resolveWith(getOutlineColor);
+
+    Widget buildSwitchListTile({required bool enabled, required bool selected}) {
+      return wrap(
+        child: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) {
```

Comment:
Do we really need a StatefulBuilder here? we do't appear to call `setState`

Reply:
Just removed all the unnecessary `StatefulBuilder`s in test.